prerace_setup.py

```diff
@@ -2,52 +2,51 @@
 
 from animal_factory import factory
 from config import RACERS_CONFIG_PATH
-from constants import *
+from constants import READ, ORIGIN0_OFFSET
 
+TRACK_LENGTH = 'track_length'
+RACERS = 'racers'
+ANIMAL_TYPE = 'type'
+RACERS_CONFIG = 'racers_config'
+RACE_SETTINGS = 'race_settings'
+ROUNDS = 'rounds'
 
-def _initialize_rounds(race_settings) -> dict:
+
+def initialize_rounds(race_settings: dict) -> dict:
     rounds = {}
     for i, curr_round in enumerate(race_settings):
         racers = _build_round_racers(curr_round, race_settings)
-        _build_round(curr_round, i, race_settings, racers, rounds)
+        rounds = _build_round(curr_round, i, race_settings, racers, rounds)
     return rounds
 
 
-def _build_round(curr_round, i, race_settings, racers, rounds):
+def load_race_settings() -> dict:
+    with open(RACERS_CONFIG_PATH, READ) as cfg_file:
+        return json.load(cfg_file)[RACE_SETTINGS]
+
+
+def unpack_round_settings(curr_round: int, rounds: dict):
+    round_settings = rounds[curr_round]
+    return round_settings[RACERS], round_settings[TRACK_LENGTH]
+
+
+def _build_round(curr_round, i, race_settings: dict, racers, rounds):
     rounds[str(i + ORIGIN0_OFFSET)] = {
         TRACK_LENGTH: race_settings[curr_round][TRACK_LENGTH],
         RACERS: racers
     }
+    return rounds
 
 
-def _build_round_racers(curr_round, race_settings):
+def _build_round_racers(curr_round: int, race_settings: dict):
     racers_settings = race_settings[curr_round][RACERS]
-    racers = []
-    for racer_name in racers_settings:
-        _build_racer(racer_name, racers, racers_settings)
+    racers = [_build_racer(racer_name, racers_settings) for racer_name in racers_settings]
     return racers
 
 
-def _build_racer(racer_name, racers, racers_settings):
+def _build_racer(racer_name: str, racers_settings: dict):
     racer = racers_settings[racer_name]
     racer_type = racer[ANIMAL_TYPE]
-    racer_properties = [racer_name] + [value for key, value in racer.items() if key != ANIMAL_TYPE]
-    racers.append(
-        factory(racer_type,
-                racer_properties))
-
-
-def _load_race_settings() -> dict:
-    with open(RACERS_CONFIG_PATH, READ) as cfg_file:
-        return json.load(cfg_file)[RACE_SETTINGS]
-
-
-def _pre_round_preparation(curr_round, rounds):
-    return _unpack_round_settings(curr_round, rounds)
-
-
-def _unpack_round_settings(curr_round, rounds):
-    round_settings = rounds[curr_round]
-    racers = round_settings[RACERS]
-    track_length = round_settings[TRACK_LENGTH]
-    return racers, track_length
+    del racer[ANIMAL_TYPE]
```

Comment:
what?

Reply:
> what?

I remove the animal type from the properties.
once I create the animal instance it's not needed because the instance itself is the type of the animal, 
so I delete that k:v from the dict.

Reply:
using `del` is a bad practice
do your logic more beautifully